core/src/test/java/org/bptlab/cepta/detect-station-delay-tests/DetectStationArrivalDelayTest.java

```diff
@@ -0,0 +1,61 @@
+package org.bptlab.cepta;
+
+import com.google.protobuf.Timestamp;
+import org.apache.flink.api.java.tuple.Tuple2;
+import org.apache.flink.streaming.api.datastream.DataStream;
+import org.apache.flink.streaming.api.datastream.DataStreamUtils;
+import org.apache.flink.streaming.api.environment.StreamExecutionEnvironment;
+import org.bptlab.cepta.models.events.train.LiveTrainDataProtos.LiveTrainData;
+import org.bptlab.cepta.models.events.train.PlannedTrainDataProtos.PlannedTrainData;
+import org.bptlab.cepta.models.events.train.TrainDelayNotificationProtos.TrainDelayNotification;
+import org.bptlab.cepta.operators.DetectStationArrivalDelay;
+import org.bptlab.cepta.providers.LiveTrainDataProvider;
+import org.bptlab.cepta.providers.PlannedTrainDataProvider;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+public class DetectStationArrivalDelayTest {
+    @Test
+    public void testDelayPositive() throws IOException {
+        AtomicBoolean pass = new AtomicBoolean(true);
```

Comment:
Why do we use an AtomicBoolean here? We should either add a comment or just use a plain and simple boolean

Reply:
We need it for the ForEach